src/room_member.cpp

```diff
@@ -56,24 +65,39 @@ void RoomMember::HandleWifiPackets(const RakNet::Packet* packet) {
     if (type == WifiPacket::PacketType::Beacon) {
         std::lock_guard<std::mutex> lock(beacon_mutex);
         beacon_queue.emplace_back(std::move(wifi_packet));
-
-        // If we have more than the max number of buffered beacons, discard the oldest one
-        if (beacon_queue.size() > MaxBeaconQueueSize) {
-            beacon_queue.pop_front();
-        }
-    } else {
-        // TODO(Subv): Add to ringbuffer for data / non-beacons
+        EmplaceBackAndCheckSize(&beacon_queue, MaxBeaconQueueSize);
+    } else {  // For now we will treat all non-beacons as data packets
+        std::lock_guard<std::mutex> lock(data_mutex);
+        EmplaceBackAndCheckSize(&data_queue, MaxDataQueueSize);
     }
 }
 
+
 std::deque<WifiPacket> RoomMember::PopWifiPackets(WifiPacket::PacketType type, const MacAddress& mac_address) {
```

Comment:
Throughout all the files, leave a space after each flow control keyword.

If (mac_address == NoPreferredMac)